packages/api/src/context.ts

```diff
@@ -1,39 +1,79 @@
-// src/server/router/context.ts
-import * as trpc from "@trpc/server";
-// import type { FetchCreateContextFnOptions } from "@trpc/server/adapters/fetch";
-import * as trpcNext from "@trpc/server/adapters/next";
-import { prisma } from "@agreeto/db";
+import type { inferAsyncReturnType } from "@trpc/server";
+import type { CreateNextContextOptions } from "@trpc/server/adapters/next";
+// import { getServerSession } from "@acme/auth";
+import { getToken, type JWTToken } from "@agreeto/auth";
+import { prisma, type User } from "@agreeto/db";
 
-/**
- * Replace this with an object if you want to pass things to createContextInner
- */
-type CreateContextOptions = Record<string, never>;
+export type CreateContextOptions = {
+  token: JWTToken | null;
+  // session: Awaited<ReturnType<typeof getServerSession>>;
+  user: User;
```

Comment:
i exported the correct types here instead of doing typescript wizard stuff with awaited returntype etc